inst/htmlwidgets/plotly.js

```diff
@@ -300,3 +272,163 @@ HTMLWidgets.widget({
   }
 
 });
+
+
+
+/**
+ * @param graphDiv The Plotly graph div
+ * @param group The crosstalk group object
+ */
+function TraceManager(graphDiv) {
+  // The Plotly graph div
+  this.gd = graphDiv;
+
+  // Preserve the original data. We'll subset based off of this whenever
+  // filtering is (re)applied.
+  this.origData = JSON.parse(JSON.stringify(graphDiv.data));
+
```

Comment:
@jcheng5 just getting back to this now after a major overhaul of `subplot()`...can't wait to dig in and help out!

First question: how come we need `JSON.parse(JSON.stringify())` here? To verify it's valid JSON?